### PR TITLE
feat: add connector props and format options in rw_catalog

### LINKED ISSUE #19072: feat(system table): add `properties map(varchar,varchar)` in rw_sources and rw_sinks
### Is your feature request related to a problem? Please describe.

Required for the source page of the cloud portal: https://linear.app/risingwave-labs/issue/CLOUD-3318/%5Bfeature-request%5D-source-details-page

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

resolve #19072

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
for `rw_catalog.rw_sources`

```
id | name | schema_id | owner | connector |      columns      |      format       |    row_encode    | append_only | associated_table_id | connection_id |                                                                                             definition                                                                                             |              acl              |      initialized_at       |        created_at         |           initialized_at_cluster_version            |             created_at_cluster_version              | is_shared |                                                                                       connector_props                                                                                       |                          format_encode_options
----+------+-----------+-------+-----------+-------------------+-------------------+------------------+-------------+---------------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------+---------------------------+---------------------------+-----------------------------------------------------+-----------------------------------------------------+-----------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------
  7 | t    |         2 |     1 | KAFKA     | {bar,foo,_row_id} | FORMAT_TYPE_PLAIN | ENCODE_TYPE_AVRO | t           |                   6 |               | CREATE TABLE t () WITH (connector = 'kafka', properties.bootstrap.server = '127.0.0.1:9092', topic = 'avro_alter_table_test') FORMAT PLAIN ENCODE AVRO (schema.registry = 'http://127.0.0.1:8081') | {postgres=r/root,root=r/root} | 2024-12-05 07:54:35+00:00 | 2024-12-05 07:54:35+00:00 | PostgreSQL 13.14.0-RisingWave-2.2.0-alpha (unknown) | PostgreSQL 13.14.0-RisingWave-2.2.0-alpha (unknown) | f         | {"connector":{"type":"plaintext","value":"kafka"},"properties.bootstrap.server":{"type":"plaintext","value":"127.0.0.1:9092"},"topic":{"type":"plaintext","value":"avro_alter_table_test"}}  | {"schema.registry":{"type":"plaintext","value":"http://127.0.0.1:8081"}}
 12 | t1   |         2 |     1 | KAFKA     | {bar,foo,_row_id} | FORMAT_TYPE_PLAIN | ENCODE_TYPE_AVRO | t           |                  11 |               | CREATE TABLE t1 () WITH (connector = 'kafka', properties.bootstrap.server = secret secret_1, topic = 'avro_alter_table_test') FORMAT PLAIN ENCODE AVRO (schema.registry = 'http://127.0.0.1:8081') | {postgres=r/root,root=r/root} | 2024-12-05 08:00:40+00:00 | 2024-12-05 08:00:40+00:00 | PostgreSQL 13.14.0-RisingWave-2.2.0-alpha (unknown) | PostgreSQL 13.14.0-RisingWave-2.2.0-alpha (unknown) | f         | {"connector":{"type":"plaintext","value":"kafka"},"properties.bootstrap.server":{"type":"secret","value":{"value":"secret_1"}},"topic":{"type":"plaintext","value":"avro_alter_table_test"}} | {"schema.registry":{"type":"plaintext","value":"http://127.0.0.1:8081"}}
(2 rows)
```

File: src/frontend/src/catalog/system_catalog/rw_catalog/rw_sinks.rs [hunk 1]
```diff
@@ -36,6 +38,11 @@ struct RwSink {
     created_at: Option<Timestamptz>,
     initialized_at_cluster_version: Option<String>,
     created_at_cluster_version: Option<String>,
+
+    // connector properties in json format
+    connector_props: String,
```
Comment (yezizp2012) [inline src/frontend/src/catalog/system_catalog/rw_catalog/rw_sinks.rs:43]: Can we directly use `JsonbVal` to enhance readability?

File: src/frontend/src/catalog/system_catalog/rw_catalog/rw_sinks.rs [hunk 2]
```diff
@@ -36,6 +38,11 @@ struct RwSink {
...
+    // format and encode properties in json format
+    format_encode_options: String,
```
Comment (yezizp2012) [inline src/frontend/src/catalog/system_catalog/rw_catalog/rw_sinks.rs:45]: Ditto.

**Review by yezizp2012 — COMMENTED** [review]
Rest LGTM

**Review by yezizp2012 — APPROVED** [review]
Rest LGTM

**Comment by fuyufjh** [comment]
Did it include authorization? i.e. Can I (non-root) see a source/table/sink created by other users?

**Comment by yezizp2012** [comment]
> Did it include authorization? i.e. Can I (non-root) see a source/table/sink created by other users?

No for most system catalogs for now. I will add uniformly. #19722

**Pull request merged**